src/balance.ts

```diff
@@ -89,6 +99,7 @@ export class Balance {
         };
     }
 
+    // TODO: We should not keep a property of a token instance (its nonce) here, in the "Tokens" (still called "Balance") class.
```

Comment:
also, I think the number of decimals has to be passed as well into this component, if we want to use it for different tokens